With UK layouts, the grave key should operate as a dead key,
and the bar should be broken.
Additionally, vowels should be given an acute accent when AltGr is pressed.

diff -uNr xkeyboard-config-1.7.orig/symbols/gb xkeyboard-config-1.7/symbols/gb
--- xkeyboard-config-1.7.orig/symbols/gb	2009-08-15 23:31:43.000000000 +0900
+++ xkeyboard-config-1.7/symbols/gb	2011-05-19 12:56:47.000000000 +0900
@@ -151,7 +151,15 @@ xkb_symbols "dvorak" {
     key <AE03> { [	    3,	sterling,  threesuperior, NoSymbol	] };
     key <BKSL> { [ numbersign,	asciitilde	] };
     key <LSGT> { [  backslash,	bar		] };
-    key <TLDE> { [	grave,	notsign,   bar,           bar		] };
+    key <TLDE> { [  deadgrave, notsign,   brokenbar,     brokenbar     ] };
+    key <AC01> { [         a,        A,   aacute,        Aacute        ] };
+    key <AC02> { [         o,        O,   oacute,        Oacute        ] };
+    key <AC03> { [         e,        E,   eacute,        Eacute        ] };
+    key <AC04> { [         u,        U,   uacute,        Uacute        ] };
+    key <AC05> { [         i,        I,   iacute,        Iacute        ] };
+    key <AD05> { [         y,        Y,   yacute,        Yacute        ] };
+    key <AD08> { [         c,        C, ccedilla,        Ccedilla      ] };
+    key <AB08> { [         w,        W,   wacute,        Wacute        ] };
 };
 
 // Dvorak letter positions, but punctuation all in the normal UK positions.
